nmt.go

```diff
@@ -41,36 +81,42 @@ type NamespacedMerkleTree struct {
 // and for the given namespace size (number of bytes).
 // If the namespace size is 0 this corresponds to a regular non-namespaced
 // Merkle tree.
-func New(h hash.Hash, namespaceSize namespace.Size) *NamespacedMerkleTree {
-	treeHasher := internal.NewNmtHasher(namespaceSize, h)
+func New(h hash.Hash, setters ...Option) *NamespacedMerkleTree {
+	// default options:
+	opts := &Options{
+		InitialCapacity:    128,
+		NamespaceIDSize:    32,
+		IgnoreMaxNamespace: false,
```

Comment:
Should we default `IgnoreMaxNamespace` to true? The only user will us for some time and we expect to use it with ignoring the parity share ID.

Also: did we settle on the 32 bytes for a namespace ID in the spec? IMO 8 or 16 bytes should be enough.
```suggestion
		NamespaceIDSize:    16,
		IgnoreMaxNamespace: true,
```

Reply:
I don't see any issue with defaulting to `true`, since the NMT is supposed to be used in conjunction with data availability proofs anyways.

For the namespace ID size, nothing is final, but you can open an issue in https://github.com/LazyLedger/lazyledger-specs to discuss tradeoffs with different sizes.

Reply:
OK, I'll set `IgnoreMaxNamespace` to true for now and leave the 32 bytes for now (as this seems to be the closest to the spec rn, e.g. given the constants and the reserved namespace IDs). I'll open an issue tomorrow :-)

Reply:
https://github.com/lazyledger/lazyledger-specs/issues/62